### PR TITLE
Access data from katdal more efficiently

### PR DESCRIPTION
**Pull request by bmerry**
- Load vis, weights and flags together (closes SPR1-359)
- Align loads to chunk boundaries, to avoid loading some chunks twice.

### CODE REVIEW

File: katsdpimager/loader_katdal.py [hunk 1]
```diff
@@ -222,15 +224,39 @@ def data_iter(self, start_channel, stop_channel, max_chunk_vis=None):
         # timestamps is a property, so ensure it's only evaluated once
         timestamps = self._file.timestamps
         baseline_idx = np.arange(len(self._baselines)).astype(np.int32)
-        for start in range(0, n_file_times, load_times):
+        start = 0
+        # Determine chunking scheme
+        if isinstance(self._file.vis, DaskLazyIndexer):
+            chunk_sizes = self._file.vis.dataset.chunks[0]
+            chunk_boundaries = [0] + list(itertools.accumulate(chunk_sizes))
```
Comment (ludwigschwardt) [inline katsdpimager/loader_katdal.py]: Why not `np.cumsum`?

File: katsdpimager/loader_katdal.py [hunk 2]
```diff
@@ -222,15 +224,39 @@ def data_iter(self, start_channel, stop_channel, max_chunk_vis=None):
...
+            if chunk_sizes and chunk_sizes[0] > load_times:
+                _logger.warning('Chunk size is %d dumps but only %d loaded at a time. '
+                                'Consider increasing --vis-load',
+                                chunk_sizes[0], load_times)
+        else:
+            chunk_boundaries = list(range(n_file_times + 1))  # No chunk info available
+        while start < n_file_times:
             end = min(n_file_times, start + load_times)
+            # Align to chunking if possible
+            aligned_end = chunk_boundaries[bisect.bisect(chunk_boundaries, end) - 1]
```
Comment (ludwigschwardt) [inline katsdpimager/loader_katdal.py]: I was going to say "why not `np.searchsorted`?" but `bisect` is actually faster, even though it is pure Python while `np.searchsorted` is compiled. Interesting! I use `searchsorted` a lot in `CategoricalData` - might be worth replacing it. Although `bisect` is best for lists while `searchsorted` wins on arrays...

File: katsdpimager/loader_katdal.py (hunk 1 above)
Comment (bmerry) [reply katsdpimager/loader_katdal.py]: It didn't seem like something that needed to be taken to numpy-land, given that there are only a few thousand entries, and it didn't seem like numpy would make the code any more readable.

File: katsdpimager/loader_katdal.py (hunk 2 above)
Comment (bmerry) [reply katsdpimager/loader_katdal.py]: bisect.bisect appears to be a built-in function.

**Pull request merged**